jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/LogStreamActor.kt

```diff
@@ -64,12 +64,33 @@ sealed class LogStreamActor(
         for (message in channel) {
             when (message) {
                 is Message.LOAD_FORWARD -> if (!nextForwardToken.isNullOrEmpty()) {
+                    withContext(edtContext) { table.setPaintBusy(true) }
                     val items = loadMore(nextForwardToken, saveForwardToken = true)
-                    withContext(edtContext) { table.listTableModel.addRows(items) }
+                    withContext(edtContext) {
+                        table.listTableModel.addRows(items)
+                        table.setPaintBusy(false)
+                    }
                 }
                 is Message.LOAD_BACKWARD -> if (!nextBackwardToken.isNullOrEmpty()) {
+                    withContext(edtContext) { table.setPaintBusy(true) }
                     val items = loadMore(nextBackwardToken, saveBackwardToken = true)
-                    withContext(edtContext) { table.listTableModel.items = items + table.listTableModel.items }
+                    if (items.isNotEmpty()) {
+                        val newSelection = table.selectedRows.map { it + items.size }
+                        table.listTableModel.items = items + table.listTableModel.items
```

Comment:
Does listTableModel.addRows(0, items) not work? Can that prevent the selection from having to be redone?

Reply:
Nope, cannot append to the front except with insertRow, let me try that out and see if it doesn't mutate selection

Reply:
This method doesn't exist..what am I thinking...

Reply:
Using insertRow fixes selection but not scrolling, the correct fix would probably be to make our own model that can handle prepending without scrolling.

Reply:
Hm well that's not great either because it requires a fork of the model (it's not extensible) and will still have scrolling issues